### DISCUSSION TITLE
webpack5 asset module - [DEP_WEBPACK_MODULE_ISSUER] DeprecationWarning: Module.issuer: Use new ModuleGraph API

### DISCUSSION THREAD

**Discussion by we125182** [webpack/webpack#16211] [Q&A]
```js
// webpack image rule
{
  test: /\.(png|jpe?g|gif|bpm|svg|webp)(\?.*)?$/,
  type: 'asset',
  parser: {
    dataUrlCondition: function (source, { module }) {
      if (/\.less|vue&type=style/.test(module.issuer.resource)) {
        return true
      }
      return source.length < 8092
    }
  },
  ...
},
```
runing the config, I get this:
```bash
(node:4765) [DEP_WEBPACK_MODULE_CHUNKS_ITERABLE] DeprecationWarning: Module.chunksIterable: Use new ChunkGraph API
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:4765) [DEP_WEBPACK_MODULE_ISSUER] DeprecationWarning: Module.issuer: Use new ModuleGraph API
(node:4765) [DEP_WEBPACK_MODULE_PROFILE] DeprecationWarning: Module.profile: Use new ModuleGraph API
```

it means need change code: `module.issuer.resource` => `compilation.moduleGraph.getIssuer(module).resource`.

but `parser.dataUrlCondition` type is `(source: Buffer, { module: Module, filename: string}) => boolean`

if there is a way to get current compilation that contain moduleGraph property?

**Comment by alexander-akait** [comment]
Here it is https://github.com/webpack/webpack/blob/main/types.d.ts#L13115 with some static functions

**Reply by we125182** [reply]
I have readed the source code of `lib/asset/AssetParser`, `lib/Module` and `lib/ModuleGraph`. but what I find is only a function  `ModuleGraph.getModuleGraphForModule` which is deprecated. can not find other way get ModuleGraph by Module.

Thank you anyway for response.
